working/augmentation-libraries/feature-specification.md

```diff
@@ -442,56 +414,142 @@ variable type, parameter type, or type parameter bound. In the last case,
+
+*In short, an augmenting declaration can omit type annotations, but if it
+doesn't, it must repeat the type from the augmented definition.*
+
+#### Inheriting combined getter setter signatures
+
+An instance getter and instance setter can be augmented with an abstract
+variable declaration because the latter is syntactic sugar for an abstract
+getter and setter declaration. This leads to a tricky edge case where the
+augmenting abstract variable may want to inherit a type but the getter and
+setter it inherits from have different types:
+
+```dart
+class C {
+  int get x => 1;
+  set x(String value) {}
+
+  @metadataToAdd
+  augment abstract var x; // What type is inherited here?
+}
+```
+
+It's a **compile-time error** if an abstract variable augments a getter and
+setter that don't have a combined signature.
 
 ## Applying augmentations
 
-An augmentation declaration _D_ is a declaration marked with the new
-built-in identifier `augment`, which makes _D_ augment a declaration _D1_
-with the same name and in the same context as _D_. _D1_ is determined as
-being before _D_ and after every other declaration with the same name and
-in the same context which is before _D_ *(that is, _D1_ is the greatest
-declaration which is smaller than _D_, according to the 'after'
-ordering)*. A compile-time error occurs if no declaration satisfies the
-requirements on _D1_.
+An augmentation declaration *D* is a declaration marked with the built-in
+identifier `augment`. We add `augment` as a built-in identifier as a language
+versioned change, to avoid breaking pre-feature code.
+
+*D* augments a declaration *I* with the same name and in the same context as
+*D*. There may be multiple augmentations in the context of *D*. More precisely,
+*I* is the declaration before *D* and after every other declaration before *D*.
 
-We say that _D1_ is the declaration which is _augmented_ by _D_.
+It's a **compile-time error** if there is no matching declaration *I*. *In other
+words, it's an error to have a declaration marked `augment` with no declaration
+to apply it to.*
 
-*Note that _D1_ can be an augmentation declaration or an introductory
-declaration.*
+We say that *I* is the declaration which is *augmented by* *D*.
+
+*In other words, take all of the declarations with the same name in some
+context, order them according to the "after" relation, and each augments the
+result of all the prior augmentations applied to the original declaration . The
+first one must not be marked `augment` and all the subsequent ones must be.*
 
 An augmentation declaration does not introduce a new name into the surrounding
 scope. *We could say that it attaches itself to the existing name.*
 
-Making `augment` a built-in identifier is language versioned, to make it
-non-breaking for pre-feature code.
+### Complete and incomplete declarations
+
+Augmentations aren't allowed to *replace* code, so they mostly add entirely new
```

Comment:
In terms of implementation complexity, it is fairly easy to support this. Actually the ability to add to the extends/with/implements clauses of a class is at a similar if not higher complexity.

Reply:
That's good to know. I'm still somewhat interested in supporting `augmented()`, but given that we have other more pressing issues, my plan is to leave it out, get this proposal cleaned up and locked down, and move on. We can always revisit augmentations and make them more powerful in the future when we see how users are using them.